### ISSUE TITLE
cluster_cells: train_batch(verbose=True) needs ~30 GB for 400k cells (OOM) => CRASH

### ISSUE THREAD

**Issue by ndaniel** [farkkilab/tribus#98]
## Summary

`cluster_cells` in `tribus/classify.py` trains the SOM with

https://github.com/farkkilab/tribus/blob/64613609f8de618c23960e09a3a5b7a026fd3ad7/tribus/classify.py#L62-L68

```python
grid_size = int(np.sqrt(np.sqrt(len(sample_data)) * 5))
...
som.train_batch(marker_data, 500, verbose=True)
```

`verbose=True` does more in MiniSom than print a progress bar. At the end of `train()` MiniSom computes the quantization error over the **whole** input (`minisom.py` line 543 in 2.3.6; the same code is in 2.3.1, which `tribus_environment.yml` pins). `quantization_error` calls `quantization`, which calls `_distance_from_weights`, and that builds one dense float64 matrix of shape `(n_cells, grid_size**2)` plus several temporaries of the same shape.

With the TRIBUS grid rule the number of nodes grows with `sqrt(n_cells)`, so the matrix grows with `n_cells ** 1.5`:

| cells | grid | nodes | one copy of the matrix (n x nodes x 8 bytes) |
| --- | --- | --- | --- |
| 50,000 | 33 x 33 | 1,089 | 0.4 GB |
| 200,000 | 47 x 47 | 2,209 | 3.5 GB |
| 415,120 | 56 x 56 | 3,136 | 10.4 GB |

Measured peak memory is about three times one copy (see reproduction). For our largest TMA slide (415,120 cells, one Tribus run per slide) the Global level needs roughly 30 GB just for this printed number. On a 39 GB laptop the run was killed three times on the same day (twice by systemd-oomd, the second time taking the desktop down, once by the kernel OOM killer inside a 16 GB cgroup cap). The six smaller slides (114k to 303k cells) passed only because their matrices happened to fit.

None of this affects the labels. The cell to node assignment is done afterwards with `som.winner(x)` one cell at a time (line 70), and the quantization error is only printed. The progress bar is also of little use: `train_batch(marker_data, 500)` performs 500 single-sample updates, so it completes in well under a second.

## Reproduction (standalone, MiniSom only, mirrors the TRIBUS call)

```python
"""usage: python3 repro.py N_CELLS {true|false|chunked}"""
import resource, sys, time
import numpy as np
from minisom import MiniSom
from numpy.linalg import norm

n = int(sys.argv[1]); mode = sys.argv[2]
rng = np.random.default_rng(42)
data = rng.standard_normal((n, 16))              # 16 z-scored markers, as TRIBUS feeds them
g = int(np.sqrt(np.sqrt(n) * 5))                 # TRIBUS grid rule
som = MiniSom(g, g, data.shape[1], sigma=1, learning_rate=1, random_seed=42)
som.random_weights_init(data)

if mode == "chunked":                            # same number, bounded memory
    def qe(self, d, chunk=20_000):
        d = np.asarray(d); tot = 0.0
        for s in range(0, d.shape[0], chunk):
            c = d[s:s + chunk]; tot += norm(c - self.quantization(c), axis=1).sum()
        return tot / d.shape[0]
    MiniSom.quantization_error = qe

t0 = time.time()
som.train_batch(data, 500, verbose=(mode in ("true", "chunked")))
peak = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss / 1024**2
print(f"n={n} grid={g}x{g} mode={mode} peak_rss={peak:.1f} GB {time.time()-t0:.0f}s")
```

## Suggested fix

One line in `cluster_cells` from `tribus/classify.py`, should have `verbose=False` instead `verbose=True`:

```python
som.train_batch(marker_data, 500, verbose=False)
```

If you want to keep the quantization error as a diagnostic, compute it in row chunks as in the reproduction (same value up to floating point summation order, memory bounded by `chunk x nodes`), or on a subsample. A chunked error computation was proposed to MiniSom in JustGlowing/minisom#96 (open since 2021, not merged), so the memory cost is on MiniSom's side too, but TRIBUS is the one asking for it on every run.

## Environment

- TRIBUS master, commit 6461360 (2025-01-20), installed with `pip install -e .`
- MiniSom 2.3.6 (the same code path exists in 2.3.1, the version pinned in `tribus_environment.yml`)
- NumPy 2.4.3, pandas 2.3.3, Python 3.12.3, Ubuntu 24.04.5, 39 GB RAM
- Input: one 16-marker logic table, one run per TMA slide of 114k to 415k cells, depth 2
